packages/accordion/src/index.js

```diff
@@ -0,0 +1,406 @@
+    index: (props, name, compName, ...rest) => {
+      if (props.index == null && props.onChange == null) {
+        return new Error(
+          "You provided an `index` prop to `Accordion` without an `onChange` handler. This will render a read-only accordion element. If the accordion should be mutable, remove `index` and use the `active` prop on the nested items that should be active. Otherwise, set `onChange`."
+        );
+      }
+      return PropTypes.number(name, props, compName, ...rest);
+    },
+    onChange: PropTypes.func,
+    readOnly: PropTypes.bool,
+    toggle: PropTypes.bool
+  };
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// AccordionItem
+
+export const AccordionItem = forwardRef(function AccordionItem(
+  { _index: index, children, disabled = false, ...props },
+  forwardedRef
+) {
+  const { accordionId, activeIndex, readOnly } = useAccordionContext();
+
+  // We need unique IDs for the panel and trigger to point to one another
+  const itemId = makeId(accordionId, index);
+  const panelId = makeId("panel", itemId);
+  const triggerId = makeId("trigger", itemId);
+
+  const context = {
+    active: activeIndex === index,
+    disabled,
+    triggerId,
+    index,
+    itemId,
+    panelId
+  };
+
+  return (
+    <AccordionItemContext.Provider value={context}>
+      <div
+        ref={forwardedRef}
+        data-reach-accordion-item=""
+        data-active={activeIndex === index ? "" : undefined}
+        data-disabled={disabled ? "" : undefined}
+        data-read-only={readOnly ? "" : undefined}
+        {...props}
+      >
+        {children}
+      </div>
+    </AccordionItemContext.Provider>
+  );
+});
+
+AccordionItem.displayName = "AccordionItem";
+if (__DEV__) {
+  AccordionItem.propTypes = {
+    disabled: PropTypes.bool
+  };
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// AccordionTrigger
+
+export const AccordionTrigger = forwardRef(function AccordionTrigger(
+  {
+    as: Comp = "button",
+    children,
+    onClick,
+    onKeyDown,
+    onMouseDown,
+    onPointerDown,
+    ...props
+  },
+  forwardedRef
+) {
+  const {
+    enabledIndices,
+    focusabledTriggerNodes,
+    onSelectPanel,
+    readOnly
+  } = useAccordionContext();
+
+  const {
+    active,
+    disabled,
+    triggerId,
+    index,
+    panelId
+  } = useAccordionItemContext();
+
+  /*
+   * If the user decides to use a div instead of a native button, we check the
+   * ref's node type after it mounts to the DOM in order to shim the necessary
+   * attributes.
+   */
+  const [isButtonElement, setIsButtonElement] = useState(Comp === "button");
+  const ownRef = useRef(null);
+  const setButtonRef = useCallback(
+    node => {
+      ownRef.current = node;
+      if (node && Comp !== "button") {
+        setIsButtonElement(node.nodeName === "BUTTON");
+      }
+    },
+    [Comp]
+  );
+  const buttonAttributeProps = isButtonElement
```

Comment:
I'd say to only pass in props for a proper button. If they want to use a different kind of element they're going to need a lot more than just a couple attributes and a keyboard handler (and I don't want to get into the business of recreating all the behaviors of buttons like we do later in this code. Either they use a button, or give us a component with the same props as a button and it's your fault for screwing up the event handlers if it doesn't work like a button)